### PR TITLE
[Feature] Change BATCH API to use a builder object instead of just an array

### PR DESCRIPTION
**Pull request by VishaalK**
After a design discussion with the team, it was decided instead of passing an array of `BatchRequest`s into a method, `batchRequest(BatchRequest[])`, it would be better to have an object that one adds `BatchRequestOperations` to. 

Now, to send a batch request, a user does the following.
```
let req = new OneNoteApi.BatchRequest();
req.addOperation({
    httpMethod: "POST",
    uri: "/api/v1.0/me/notes"
    contentType: "text/html"
    content: "<html><head></head><body></body></html>"
} as OneNoteApi.BatchRequestOperation);

return api.sendBatchRequest(req).then((response) => { console.log("profit"); });
```

### CODE REVIEW

File: src/scripts/batchRequest.ts [hunk 1]
```diff
@@ -0,0 +1,68 @@
...
+
+	public getNumOperations(): number {
+		return this.operations.length;
+	}
+
+	public getRequestBody(): string {
+		return this.convertOperationsToHttpRequestBody();
+	}
+
+	private convertOperationsToHttpRequestBody(): string {
```
Comment (mttwc) [inline src/scripts/batchRequest.ts]: getRequestBody() simply returns convertOperationsToHttpRequestBody(). You could remove the private method and move its body into getRequestBody()

File: src/scripts/batchRequest.ts [hunk 2]
```diff
@@ -0,0 +1,68 @@
...
+
+	public addOperation(op: BatchRequestOperation): void {
+		this.operations.push(op);
+	}
+
+	public getOperation(index: number): BatchRequestOperation {
+		return this.operations[index];
+	}
+
+	public getOperations(): BatchRequestOperation[] {
```
Comment (mttwc) [inline src/scripts/batchRequest.ts]: I'd avoid exposing this.operations in this way as any modifications made to the list will also modify the BatchRequest object. The get() and getNumOperations() should be sufficient.

**Review by mttwc — CHANGES_REQUESTED** [review]

File: src/scripts/batchRequest.ts (hunk 1 above)
Comment (VishaalK) [reply src/scripts/batchRequest.ts]: Will do!

File: src/scripts/batchRequest.ts (hunk 2 above)
Comment (VishaalK) [reply src/scripts/batchRequest.ts]: That's fair. If we decide to change the underlying implementation (which we eventually will I am sure, to something like a map) then this method will break. Will fix now.

**Comment by VishaalK** [comment]
All things are done!

**Review by mttwc — APPROVED** [review]

**Comment by mttwc** [comment]
Looks good!

**Pull request merged**